mitiq/zne/scaling/tests/test_folding.py

```diff
@@ -960,19 +960,94 @@ def test_global_fold_stretch_factor_eight_terminal_measurements():
         circ,
         inverse(circ),
         circ,
-        inverse(
-            Circuit(
-                [ops.T.on(qreg[2])],
-                [ops.TOFFOLI.on(*qreg)],
-            )
-        ),
+        inverse(circ)[0],
+        circ[-1],
+        meas,
+    )
+    assert _equal(folded, correct)
```

Comment:
@andreamari I have attempted a quick fix for the input circuit below but that caused an issue with an already existing test. The global folded circuit that we get back is a bit different compared to how the original global folding function would insert gates etc. 

Will spend some time trying to understand how to get the intended folded circuit back but wanted to add this comment in case you catch my mistake quickly. 

Input circuit:
0: ───H───────────────

1: ───────Z───────────

2: ───────────S───────

3: ───────────────T───